_code/InSituTree/R/supervisedSubcluster.R

```diff
@@ -0,0 +1,97 @@
+#' Supervised subclustering using a gene subset and insitutypeML
+#'
+#' @param reference_profiles Matrix of expression profiles, genes x cell types
+#' @param x Counts matrix, cells x genes
+#' @param neg Vector of mean negative controls for each cell
+#' @param bg Expected background
+#' @param quantile_absolute_expression_difference Minimum absolute expression
+#' difference within reference_profiles in terms of quantile level among all
+#' genes. Default = 0.5. Both quantile cutoffs must be passed to retain gene.
+#' @param quantile_percent_expression_difference Minimum percentage expression
+#' difference within reference_profiles in terms of quantile level among all
+#' genes. Default = 0.5. Both quantile cutoffs must be passed to retain gene.
+#' @param excluded_genes Genes to be excluded during fitting with InSituType
+#' @param cohort Vector of cells' cohort membership
+#'
+#' @import InSituType
+#'
+#' @return a list with all elements returned by InSituType::insitutypeML() and
+#' additional element `ctsPerCell` for a vector of counts per cell within the
+#' chosen subset of genes.
+#' @export
+
+supervisedSubcluster <- function(reference_profiles,
+                                 x,
+                                 neg,
```

Comment:
should both `neg` and `bg` be required for this function? if not, you can set one of them to `NULL`, say `neg = NULL`.  
Alternatively, is to force your function to use `bg` instead of `neg` since you would always have a not NULL `bg` value from the `runInSituTree()` function. In this case, it would be better to clarify the parameter description for `bg` as the "vector of expected background for each cell after background estimation".

Reply:
Yes, `neg` is needed for the `estep` of `insitutypeML` and throws a warning if it's missing. As you pointed out, it will not be used for background estimation in the cell typing since `bg` is also provided

Reply:
Description for `bg` was taken directly from `insitutypeML`, but I'll add this clarification you have above.